diff --git a/drivers/media/platform/mxc/capture/adv7610.c b/drivers/media/platform/mxc/capture/adv7610.c
index b6d7246..1fba4c2 100644
--- a/drivers/media/platform/mxc/capture/adv7610.c
+++ b/drivers/media/platform/mxc/capture/adv7610.c
@@ -47,7 +47,7 @@
 #include "mxc_v4l2_capture.h"
 
 #define USE_16BIT
-//#define SUPPORT_1080P
+#define SUPPORT_1080P
 
 /*!
  * Maintains the information on the current state of the sensor.
